packages/flame/lib/src/components/position_component.dart

```diff
@@ -1,218 +1,349 @@
+  Vector2 get absoluteCenter => absolutePositionOfAnchor(Anchor.center);
 
-  /// Angle (with respect to the x-axis) this component should be rendered with.
-  /// It is rotated around its anchor.
-  double angle;
+  //#endregion
 
-  /// Anchor point for this component. This is where flame "grabs it".
-  /// The [position] is relative to this point inside the component.
-  /// The [angle] is rotated around this point.
-  Anchor anchor;
+  //#region Mutators
 
-  /// Whether this component should be flipped on the X axis before being rendered.
-  bool renderFlipX = false;
+  /// Flip the component horizontally around its anchor point.
+  void flipHorizontally() => _transform.flipHorizontally();
 
-  /// Whether this component should be flipped ofn the Y axis before being rendered.
-  bool renderFlipY = false;
+  /// Flip the component vertically around its anchor point.
+  void flipVertically() => _transform.flipVertically();
 
-  PositionComponent({
-    Vector2? position,
-    Vector2? size,
-    Vector2? scale,
-    this.angle = 0.0,
-    this.anchor = Anchor.topLeft,
-    this.renderFlipX = false,
-    this.renderFlipY = false,
-    int? priority,
-  })  : _position = position ?? Vector2.zero(),
-        _size = size ?? Vector2.zero(),
-        _scale = scale ?? Vector2.all(1.0),
-        super(priority: priority);
+  /// Flip the component horizontally around its center line.
+  void flipHorizontallyAroundCenter() {
+    final delta = (1 - 2 * _anchor.x) * width;
+    _transform.x += delta * math.cos(_transform.angle);
+    _transform.y += delta * math.sin(_transform.angle);
+    _transform.flipHorizontally();
+  }
 
-  @override
-  bool containsPoint(Vector2 point) {
-    final rectangle = Rectangle.fromRect(toAbsoluteRect(), angle: angle)
-      ..position = absoluteCenter;
-    return rectangle.containsPoint(point);
+  /// Flip the component vertically around its center line.
+  void flipVerticallyAroundCenter() {
+    final delta = (1 - 2 * _anchor.y) * height;
+    _transform.x += -delta * math.sin(_transform.angle);
+    _transform.y += delta * math.cos(_transform.angle);
+    _transform.flipVertically();
   }
 
-  double angleTo(PositionComponent c) => position.angleTo(c.position);
+  //#endregion
 
-  double distance(PositionComponent c) => position.distanceTo(c.position);
+  /// Internal handler which is called automatically whenever either
+  /// the [size] or [scale] change.
+  void _onModifiedSizeOrScale() {
+    _scaledSize.x = _size.x * scale.x.abs();
+    _scaledSize.y = _size.y * scale.y.abs();
+  }
+
+  /// Internal handler that must be invoked whenever either the [size]
+  /// or the [anchor] change.
+  void _onModifiedSizeOrAnchor() {
+    _transform.offset = Vector2(-_anchor.x * _size.x, -_anchor.y * _size.y);
+  }
 
   @override
   void renderDebugMode(Canvas canvas) {
-    if (this is Hitbox) {
-      (this as Hitbox).renderHitboxes(canvas);
+    super.renderDebugMode(canvas);
```

Comment:
just a comment for a potential followup, but we should find a way to extract all this debug related code into a separated class, to improve readability